ironfist/src/cpp/adventure/adv.cpp

```diff
@@ -2,6 +2,62 @@
 #include "adventure/map.h"
 #include "game/game.h"
 #include "scripting/hook.h"
+#include "prefs.h"
+
+extern void __fastcall NormalDialog(char *, int, int, int, int, int, int, int, int, int);
+
+const int BUTTON_CODE_OKAY = 30725;
+const int BUTTON_CODE_CANCEL = 30726;
+
+extern long __fastcall KBTickCount();
+
+int advManager::ProcessDeSelect(struct tag_message *GUIMessage_evt, int *a3, class mapCell * *a4) {
+	extern int giBottomViewOverride;
+	extern int iCurBottomView;
+	extern int giBottomViewOverrideEndTime;
+	
+	if(GUIMessage_evt->yCoordOrFieldID == 4) {

```

Comment:
This is one magic number to definitely remove -- presumably, 4 is the "end-turn" button. You can use h2guiview to find out for sure.